### PR TITLE
[CDAP-3051] Add documentation and an example for dataset management in programs

### PR DESCRIPTION
**Pull request by anew**
- added dataset admin endpoints to FileSetExample
- added documentation for dataset admin in developer guide

Passes [Quick Build 5](http://builds.cask.co/browse/CDAP-DOB176-4)

Pages of interest:

- [Developers’ Guide: Dataset Admin](http://builds.cask.co/artifact/CDAP-DOB176/shared/build-5/Docs-HTML/3.4.0-SNAPSHOT/en/developers-manual/building-blocks/datasets/index.html)
- [Examples: FileSet](http://builds.cask.co/artifact/CDAP-DOB176/shared/build-5/Docs-HTML/3.4.0-SNAPSHOT/en/examples-manual/examples/fileset-example.html)

### CODE REVIEW

**Comment by awholegunch** [comment]
Docs LGTM, but @anew needs to check my last changes...

File: cdap-docs/examples-manual/source/examples/fileset-example.rst [hunk 1]
```diff
@@ -125,8 +125,9 @@ This is done by making a RESTful call to the *FileSetService*.
 A sample text file (``lines.txt``) is included in the ``resources`` directory of the example:
 
 .. tabbed-parsed-literal::
+    :tabs: "CDAP CLI"
 
-  cdap> call service FileSetExample.FileSetService PUT "lines?path=some.txt" body:file examples/FileSetExample/resources/lines.txt
+    |cdap >| call service FileSetExample.FileSetService PUT "lines?path=some.txt" body:file examples/FileSetExample/resources/lines.txt
```
Comment (anew) [inline cdap-docs/examples-manual/source/examples/fileset-example.rst]: we actually don't need the quotes here.

**Comment by anew** [comment]
build is here: http://builds.cask.co/browse/CDAP-DUT3993

**Comment by poornachandra** [comment]
Java code LGTM

**Comment by anew** [comment]
thanks poorna, waiting for build before I will squash and merge

**Comment by anew** [comment]
build is green. Squashing and merging.

**Pull request merged**